crates/turborepo-lib/src/run/scope/filter.rs

```diff
@@ -0,0 +1,48 @@
+use tracing::debug;
+use turbopath::{AbsoluteSystemPath, AnchoredSystemPathBuf};
+
+use crate::package_graph;
+
+pub struct PackageInference {
+    package_name: Option<String>,
+    directory_root: AnchoredSystemPathBuf,
+}
+
+impl PackageInference {
+    pub fn calculate(
+        turbo_root: &AbsoluteSystemPath,
+        pkg_inference_path: &AnchoredSystemPathBuf,
+        pkg_graph: &package_graph::PackageGraph,
+    ) -> Self {
+        debug!(
+            "Using {} as a basis for selecting pacakges",
+            pkg_inference_path
+        );
+        let full_inference_path = turbo_root.resolve(pkg_inference_path);
+        for (workspace_name, workspace_entry) in pkg_graph.workspaces() {
+            let pkg_path = turbo_root.resolve(&workspace_entry.package_json_path());
+            let inferred_path_is_below = pkg_path.contains(&full_inference_path);
+            // We skip over the root package as the inferred path will always be below it
+            if inferred_path_is_below && (&pkg_path as &AbsoluteSystemPath) != turbo_root {
+                // set both. The user might have set a parent directory filter,
+                // in which case we *should* fail to find any packages, but we should
+                // do so in a consistent manner
+                return Self {
+                    package_name: Some(workspace_name.to_string()),
+                    directory_root: workspace_entry.package_json_path().clone(),
+                };
+            }
+            let inferred_path_is_between_root_and_pkg = full_inference_path.contains(&pkg_path);
```

Comment:
So if `turbo` is executed in say, `apps/` where there are `apps/docs` and `apps/web` workspaces, it'll only run tasks in these workspaces? Do we have this documented?

Reply:
[This](https://turbo.build/repo/docs/core-concepts/monorepos/filtering#via-global-turbo) is what we document. But yes, running in `apps/` will pick up `apps/docs` and `apps/web`. It is equivalent to setting `--filter={./apps/*}`